src/parse-mockdb.js

```diff
@@ -104,14 +104,30 @@ function runHook(className, hookType, data) {
     const modelData = Object.assign(new Object, data, {className});
     const model = Parse.Object.fromJSON(modelData);
 
-    return hook.bind(model)().then((result) => {
-      debugPrint('HOOK', result);
-      return Parse.Promise.as(result.toJSON());
+    // TODO Stub out Parse.Cloud.useMasterKey() so that we can report the correct 'master'
+    // value here.
+    return hook(makeRequestObject(model, false)).done((beforeSaveOverrideValue) => {
+      debugPrint('HOOK', { beforeSaveOverrideValue });
+      // Unlike BeforeDeleteResponse, BeforeSaveResponse might specify
+      var objectToProceedWith = hookType === 'beforeSave' && beforeSaveOverrideValue
```

Comment:
ok, one last nit, because the ternary has two conditions, it might be easier to read like this:

```js
var objectToProceedWith = model;
if (hookType === 'beforeSave' && beforeSaveOverrideValue) {
  objectToProceedWith = beforeSaveOverrerideValue.toJSON()
}
```

What do you think?